nextest-runner/src/platform.rs

```diff
@@ -50,19 +86,169 @@ impl BuildPlatforms {
         }
     }
 
-    /// Converts a target triple to a `String` that can be stored in the build-metadata.
+    /// Converts a target triple to a [`String`] that can be stored in the build-metadata.
     ///
-    /// cargo-nextest represents the host triple with `None` during runtime. However the
-    /// build-metadata might be used on a system with a different host triple. Therefore, the host
```

Comment:
@sunshowers If `BuildPlatformsSummary` will be used on a different machine, does that mean we shouldn't serialize the target libdir and the host libdr to the summary and we should always detect both libdirs when creating the `BuildPlatforms` from `BuildPlatformsSummary`?

Reply:
Ahh, good question. I don't expect this to be tackled in this PR, to be clear. But there are two broad strategies:

1. Go all-in: add the libdirs to the archive and use those rather than the ones on disk.
2. Discover the libdirs from the system while running them.

Given that we support running tests out of an archive without a Rust installation being present at all, I think overall 1 is probably the better approach -- if we can slim down the set of files we serialize to just the ones needed by proc-macro tests. But we may consider adding optional support for 2 as well.

I think we should still serialize these directories in the summary, but be intelligent about it while deserializing them.

Reply:
Done with approach 1.